### DISCUSSION TITLE
Required and optional support for on-disk file formats

### DISCUSSION THREAD

**Discussion by lmoureaux** [hep-statistics-standard/hep-statistics-standard#23] [Technical]
HS3 positions itself as supporting many JSON-like formats. I think JSON support should be required by the spec. This can be done by defining HS3 as a JSON dialect, then allowing other representations (which also allows the use of JSON terminology - related to #15).

Proposed wording:

HS3 documents are encoded in the JSON format as defined in ISO/IEC 21778:2017 [ref]. Implementations MAY support other serialization formats that support a non-ambiguous mapping to JSON, such as TOML [ref] or YAML [ref], in which case they SHOULD use a different file extension.

There is some interplay between this issue and #16/#17.

**Comment by cburgard** [comment]
Sounds like a great plan to me, unless anyone has a different suggestion, I think we should go with this!

**Comment by jonas-eschle** [comment]
I think there is still one elephant in the room and that is the interplay with possible binary formats: large histograms and unbinned data (be it in the likelihood or as a template - say KDE - in the model), which would need some addition. The most straightforward probably (as discussed also at PyHEP.dev) could be to have references that point into an external buffer.

But for the JSON part, I agree on the language

**Comment by cburgard** [comment]
The suggested wording has been added to the current draft.
